financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsRepository.kt

```diff
@@ -40,4 +41,24 @@ internal interface FinancialConnectionsRepository {
         clientSecret: String,
         institutionId: String
     ): FinancialConnectionsSessionManifest.FinancialConnectionsAuthorizationSession
+
+    @Throws(
+        AuthenticationException::class,
+        InvalidRequestException::class,
+        APIConnectionException::class,
+        APIException::class
+    )
+    suspend fun completeAuthorizationSession(
+        clientSecret: String,
+        sessionId: String,
+        /**
+         * MX-specific.
```

Comment:
Is it just MX specific?

I think this may be a `grep` that finds these cases but not 100% sure:
- https://livegrep.corp.stripe.com/search/stripe?q=props.onSuccess(%20path%3APartnerAuth&fold_case=auto&regex=false&context=true

Reply:
🤔  I think for the `/authorize` endpoint, the actual `public_token` parameter just applies for MX?

https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/54e59cb8/src/payment_intent/inner/instant_debits_app/InstantDebitsApp.tsx#L114

completing the auth session with Finicity did not require any extra parameter.

Reply:
Update - made this more generic after polling for OAuth results:

https://github.com/stripe/stripe-android/pull/5320/commits/24a566c9e3963a5f185b2a51e08e1263c8b7acbe

1.  When browser closes, poll `connections/auth_sessions/oauth_results`
2.  When the above succeeds, it returns a response with the shape of the screenshot below.
3. After that, call `connections/auth_sessions/$authSessionId/authorized`. This endpoint accepts an optional public_key . For MX flows at least, we need to pass the `member_guid` field received in (1). I'm passing it everytime we don't get a null value.
4. The above should succeed, and we can start polling accounts.

![image](https://user-images.githubusercontent.com/99293320/181476323-69f6ea1f-69cb-4964-bcbc-96e2a7ef342f.png)

Reply:
Yea, I don't see it required for Finicity either. WellsFargo/TestMode/TrueLayer (not sure if that still is used) may require it as per grep:
- https://livegrep.corp.stripe.com/search/stripe?q=props.onSuccess(%20path%3APartnerAuth&fold_case=auto&regex=false&context=true

Reply:
Made the assumption of `if member_guid is present, send it as public_token in the authorized endpoint`. Tested with Testmode, Finicity and MX so far, and 👌 .